pkg/controllermanager/controller/controllerinstallation/controllerinstallation_control.go

```
@@ -0,0 +1,402 @@
+// to update the status of ControllerInstallations. You should use an instance returned from NewDefaultControllerInstallationControl() for any
+// scenario other than testing.
+func NewDefaultControllerInstallationControl(k8sGardenClient kubernetes.Interface, k8sGardenInformers gardeninformers.SharedInformerFactory, k8sGardenCoreInformers gardencoreinformers.SharedInformerFactory, recorder record.EventRecorder, config *config.ControllerManagerConfiguration, seedLister gardenlisters.SeedLister, controllerRegistrationLister gardencorelisters.ControllerRegistrationLister, controllerInstallationLister gardencorelisters.ControllerInstallationLister) ControlInterface {
+	return &defaultControllerInstallationControl{k8sGardenClient, k8sGardenInformers, k8sGardenCoreInformers, recorder, config, seedLister, controllerRegistrationLister, controllerInstallationLister}
+}
+
+type defaultControllerInstallationControl struct {
+	k8sGardenClient              kubernetes.Interface
+	k8sGardenInformers           gardeninformers.SharedInformerFactory
+	k8sGardenCoreInformers       gardencoreinformers.SharedInformerFactory
+	recorder                     record.EventRecorder
+	config                       *config.ControllerManagerConfiguration
+	seedLister                   gardenlisters.SeedLister
+	controllerRegistrationLister gardencorelisters.ControllerRegistrationLister
+	controllerInstallationLister gardencorelisters.ControllerInstallationLister
+}
+
+func (c *defaultControllerInstallationControl) Reconcile(obj *gardencorev1alpha1.ControllerInstallation) error {
+	var (
+		controllerInstallation = obj.DeepCopy()
+		logger                 = logger.NewFieldLogger(logger.Logger, "controllerinstallation", controllerInstallation.Name)
+	)
+
+	logger.Infof("[CONTROLLERINSTALLATION RECONCILE] %s", controllerInstallation.Name)
+
+	if isResponsible, err := c.isResponsible(controllerInstallation); !isResponsible {
```

error is not checked

---

nice catch